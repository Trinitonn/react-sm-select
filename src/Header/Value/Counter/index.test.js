import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Counter } from './index';

describe('Counter', () => {
  it('should render properly', () => {
    const wrapper = shallow(
      <Counter
        valuePlaceholder="Some Placeholder"
        counterLabel="Counter Label"
        value={['one']}
        options={[{ value: 'one', label: 'One' }, { value: 'two', label: 'Two' }]}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render properly if all selected', () => {
    const wrapper = shallow(
      <Counter
        valuePlaceholder="Some Placeholder"
        counterLabel="Counter Label"
        value={['one', 'two']}
        options={[{ value: 'one', label: 'One' }, { value: 'two', label: 'Two' }]}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render properly if no counterLabel', () => {
    const wrapper = shallow(
      <Counter
        valuePlaceholder="Some Placeholder"
        value={['one', 'two']}
        options={[{ value: 'one', label: 'One' }, { value: 'two', label: 'Two' }]}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
