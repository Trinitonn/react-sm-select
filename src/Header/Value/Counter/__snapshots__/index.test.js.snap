// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render properly 1`] = `
<span>
  Counter Label
  <span
    className="Header__valueCounter"
  >
    (
    1
    )
  </span>
</span>
`;

exports[`Counter should render properly if all selected 1`] = `
<span>
  Counter Label
  <span
    className="Header__valueCounter"
  >
    (
    All
    )
  </span>
</span>
`;

exports[`Counter should render properly if no counterLabel 1`] = `
<span>
  Some Placeholder
  <span
    className="Header__valueCounter"
  >
    (
    All
    )
  </span>
</span>
`;
